server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -248,88 +221,72 @@ public HypervisorUpdateResultDTO hypervisorUpdate(
-        "consumers already exist it is probably best not to allow creation of new hypervisor" +
-        " consumers.  Most consumers do not have a hypervisorId attribute, so that should be" +
-        " added manually when necessary by the management environment. Default is true.  " +
-        "If false is specified, hypervisorIds that are not found will result in a failed " +
-        "state of the job.", value = "hypervisorUpdateAsync")
-    @ApiResponses({ @ApiResponse(code = 202, message = "") })
-    @POST
-    @Consumes(MediaType.TEXT_PLAIN)
-    @Produces(MediaType.APPLICATION_JSON)
-    @Path("/{owner}")
+    @Override
     @UpdateConsumerCheckIn
     @SuppressWarnings("checkstyle:indentation")
     public AsyncJobStatusDTO hypervisorUpdateAsync(
-        String hypervisorJson, @Context Principal principal,
-        @PathParam("owner") @Verify(value = Owner.class,
-            require = Access.READ_ONLY,
-            subResource = SubResource.HYPERVISOR) String ownerKey,
-        @ApiParam("specify whether or not to create missing hypervisors." +
-            "Default is true.  If false is specified, hypervisorIds that are not found" +
-            "will result in failed entries in the resulting HypervisorCheckInResult")
-
-        @QueryParam("create_missing") @DefaultValue("true") boolean createMissing,
-        @QueryParam("reporter_id") String reporterId) throws JobException {
+        @Verify(value = Owner.class, require = Access.READ_ONLY,
+        subResource = SubResource.HYPERVISOR) String ownerKey,
+        Boolean createMissing, String reporterId, String hypervisorJson) {
 
         if (hypervisorJson == null || hypervisorJson.isEmpty()) {
             log.debug("Host/Guest mapping provided during hypervisor update was null.");
             throw new BadRequestException(
                 i18n.tr("Host to guest mapping was not provided for hypervisor update."));
         }
 
+        Principal principal = (this.principalProvider == null ? null : this.principalProvider.get());
+
         log.info("Hypervisor update by principal: {}", principal);
         Owner owner = this.getOwner(ownerKey);
 
         JobConfig config = HypervisorUpdateJob.createJobConfig()
             .setOwner(owner)
             .setData(hypervisorJson)
-            .setCreateMissing(createMissing)
+            .setCreateMissing(createMissing.booleanValue())
             .setPrincipal(principal)
             .setReporter(reporterId);
 
-        AsyncJobStatus status = jobManager.queueJob(config);
-        return translator.translate(status, AsyncJobStatusDTO.class);
+        try {
+            AsyncJobStatus status = jobManager.queueJob(config);
+            return translator.translate(status, AsyncJobStatusDTO.class);
+        }
+        catch (JobException e) {
+            String errmsg = this.i18n.tr("An unexpected exception occurred while scheduling job \"{0}\"",
+                config.getJobKey());
+
+            throw new IseException(errmsg, e);
```

Comment:
What prompted this change? We didn't catch the exception beforehand
(same question applies about the exception catching in the heartbeat endpoint)

Reply:
Previously the method was throwing an unhandled `JobException`. But in auto-generated `HypervisorsApi` the method does not `throws JobException`. Hence the inherited class method is not allowing to throw an exception. Hence needed to handle and re-throw an exception. 

I didn't find any valid property that could be added in the YAML specification file which adds  `throws Exception` to an auto-generated method. The issue link says its an enhancement: https://github.com/OpenAPITools/openapi-generator/issues/4815.

Reply:
Alright. But we should log this as an error so that the full stacktrace will be available in the log. With the current way of just throwing an IseException, the error message is returned to the client, but there is nothing in the log, so we wouldn't know where to start if we were to investigate an error. Adding `log.error(errmsg, e)'` right before we throw the IseException should give us the full stacktrace in the candlepin.log

Reply:
Yes. Added error log.